src/main/java/frc/robot/commands/DeepClimbPrep.java

```diff
@@ -1,16 +1,41 @@
 package frc.robot.commands;
+import java.util.Optional;
+
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.wpilibj2.command.ConditionalCommand;
 import edu.wpi.first.wpilibj2.command.InstantCommand;
 import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
 import edu.wpi.first.wpilibj2.command.WaitUntilCommand;
 import frc.robot.subsystems.climb.Climb;
+import frc.robot.subsystems.IntakeLimelight.IntakeLimelight;
+import frc.robot.subsystems.arm.Arm;
+import frc.robot.subsystems.arm.Arm.ArmPosition;
 import frc.robot.subsystems.climb.Climb.ClimbPosition;
+import frc.robot.subsystems.elevator.Elevator;
+import frc.robot.subsystems.elevator.Elevator.ElevatorHeight;
+import edu.wpi.first.math.geometry.Transform2d;
 
 public class DeepClimbPrep extends SequentialCommandGroup {
-    public DeepClimbPrep(Climb climb) {
+    private Optional<Transform2d> isTag = Optional.empty();
+    public DeepClimbPrep(Climb climb, IntakeLimelight intakeLime, Elevator elevator, Arm arm) {
       addRequirements(climb);
-      // TODO use apriltag to allign with cage
+      isTag = intakeLime.checkForTag();
```

Comment:
if sees tag, needs to ensure the tag is for deep climb. see the method "validIntakingTag" in IntakeLimelight. make a "validDeepClimbTag" method in IntakeLimelight and use it here